drivers/dma/dma_imx_rt.c

```diff
@@ -0,0 +1,498 @@
+	if (config->dma_callback) {
+		LOG_DBG("INSTALL call back on channel %d", channel);
+		data->callback_arg = config->callback_arg;
+		data->dma_callback = config->dma_callback;
+	}
+
+	irq_unlock(key);
+
+	return 0;
+}
+
+static int dma_imx_rt_start(struct device *dev, u32_t channel)
+{
+	struct call_back *data = DEV_CHANNEL_DATA(dev, channel);
+
+	LOG_DBG("START TRANSFER");
+	LOG_DBG("DMAMUX CHCFG 0x%x", DEV_DMAMUX_BASE(dev)->CHCFG[channel]);
+	LOG_DBG("DMA CR 0x%x", DEV_BASE(dev)->CR);
+	data->busy = true;
+	EDMA_StartTransfer(DEV_EDMA_HANDLE(dev, channel));
+	return 0;
+}
+
+static int dma_imx_rt_stop(struct device *dev, u32_t channel)
+{
+	struct dma_imx_rt_data *data = DEV_DATA(dev);
+
+	if (!data->data_cb[channel].busy) {
+		return 0;
+	}
+	EDMA_AbortTransfer(DEV_EDMA_HANDLE(dev, channel));
+	EDMA_ClearChannelStatusFlags(DEV_BASE(dev), channel,
+				     kEDMA_DoneFlag | kEDMA_ErrorFlag |
+					     kEDMA_InterruptFlag);
+	EDMA_ResetChannel(DEV_BASE(dev), channel);
+	data->data_cb[channel].busy = false;
+	return 0;
+}
+
+static int dma_imx_rt_reload(struct device *dev, u32_t channel, u32_t src,
+			     u32_t dst, size_t size)
+{
+	struct call_back *data = DEV_CHANNEL_DATA(dev, channel);
+
+	if (data->busy) {
+		EDMA_AbortTransfer(DEV_EDMA_HANDLE(dev, channel));
+	}
+	return 0;
+}
+
+static int dma_imx_rt_get_status(struct device *dev, u32_t channel,
+				 struct dma_status *status)
+{
+	edma_tcd_t *tcdRegs;
+
+	if (DEV_CHANNEL_DATA(dev, channel)->busy) {
+		status->busy = true;
+		status->pending_length =
+			EDMA_GetRemainingMajorLoopCount(DEV_BASE(dev), channel);
+	} else {
+		status->busy = false;
+		status->pending_length = 0;
+	}
+	status->dir = DEV_CHANNEL_DATA(dev, channel)->dir;
+	LOG_DBG("DMAMUX CHCFG 0x%x", DEV_DMAMUX_BASE(dev)->CHCFG[channel]);
+	LOG_DBG("DMA CR 0x%x", DEV_BASE(dev)->CR);
+	LOG_DBG("DMA INT 0x%x", DEV_BASE(dev)->INT);
+	LOG_DBG("DMA ERQ 0x%x", DEV_BASE(dev)->ERQ);
+	LOG_DBG("DMA ES 0x%x", DEV_BASE(dev)->ES);
+	LOG_DBG("DMA ERR 0x%x", DEV_BASE(dev)->ERR);
+	LOG_DBG("DMA HRS 0x%x", DEV_BASE(dev)->HRS);
+	tcdRegs = (edma_tcd_t *)((uint32_t)&DEV_BASE(dev)->TCD[channel]);
+	LOG_DBG("data csr is 0x%x", tcdRegs->CSR);
+	return 0;
+}
+
+static const struct dma_driver_api dma_imx_rt_api = {
+	.reload = dma_imx_rt_reload,
+	.config = dma_imx_rt_configure,
+	.start = dma_imx_rt_start,
+	.stop = dma_imx_rt_stop,
+	.get_status = dma_imx_rt_get_status,
+};
+
+static int dma_imx_rt_init(struct device *dev)
+{
+	edma_config_t userConfig = { 0 };
+
+	LOG_DBG("INIT NXP DMA");
+	DMAMUX_Init(DEV_DMAMUX_BASE(dev));
+	EDMA_GetDefaultConfig(&userConfig);
+	EDMA_Init(DEV_BASE(dev), &userConfig);
+	DEV_CFG(dev)->irq_config_func(dev);
+	memset(DEV_DATA(dev), 0, sizeof(struct dma_imx_rt_data));
+	memset(tcdpool, 0, sizeof(tcdpool));
+	return 0;
+}
+
+static void dma_imx_config_func_0(struct device *dev);
+
+static const struct dma_imx_rt_config dma_config_0 = {
+	.base = (DMA_Type *)DT_INST_0_NXP_IMX_EDMA_BASE_ADDRESS_0,
```

Comment:
Please use the new DT_INST macros as in commit 4b9fe8a24e76b77e157c4d0d50f8c6aabadd9e87

Reply:
ok